lte/gateway/c/session_manager/SessionStateEnforcer.h

```diff
@@ -159,6 +160,24 @@ class SessionStateEnforcer {
       std::unique_ptr<SessionState>& session, SessionFsmState target_state,
       SessionStateUpdateCriteria* session_uc);
 
+  /*
+   * Acquire and update session rules based on the IMSI
+   *
+   * */
+  uint32_t update_session_rules(
+      const std::string& imsi, std::unique_ptr<SessionState>& session_state,

```

Comment:
Do we need to pass in imsi as a parameter in all these functions? It should be easy to get it from `session_state->get_imsi()`

Reply:
removed the parameter imsi in few places wherever possible for this PR